plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt

```diff
@@ -294,11 +301,33 @@ class ScaleBarImpl : ScaleBar, View {
     }
   }
 
-  private fun drawText(canvas: Canvas, text: String, x: Float, y: Float) {
+  private fun drawText(canvas: Canvas, text: String, x: Float, y: Float, lastUnitBarText: Boolean) {
+    var safeX = x
+
+    // As an optimization only check if it goes beyond right view for the last unit bar text
+    if (lastUnitBarText) {
+      // Check if it goes beyond the right margin of the view
+      val textWidthPx = textPaint.measureText(text)
+      val textMaxRightPx = x + strokePaint.strokeWidth + when (textPaint.textAlign) {
+        Paint.Align.LEFT -> textWidthPx
+        Paint.Align.CENTER -> textWidthPx / 2
+        Paint.Align.RIGHT, null -> 0F
+      }
+      if (textMaxRightPx > width) {
+        // Move it away from right margin enough to fit
+        safeX -= textMaxRightPx - width
+      }
+    }
+
+    // Check if it goes beyond the left margin of the view
+    if (x - strokePaint.strokeWidth / 2 < 0) {
```

Comment:
is it ok that we check `x` here but operate with `safeX` below?

Reply:
You're right. It's better to use `safeX`.

Reply:
and also for better readability I'd use `if (x < trokePaint.strokeWidth / 2)` similar to L316

Reply:
Hmmm I personally think that:
```Kotlin
if (x < trokePaint.strokeWidth / 2)
```
it's a bit obscure. 
I need to think more to figure out why we're comparing a position (`x` or `safeX`) with a width (`strokeWidth`). Somehow, to me, mine reads easier: to position `safeX` I remove a width and then check if it goes beyond the left minimum position (`0`).

But happy to change it to your proposal if others also prefer it.